wordpress-mobile/WordPress-Android
# [Bloganuary] Create Learn More nudge modal UI

Part of #19685 

Create the modal UI and basic UI-related actions (such as modal dismiss by user). This still doesn't have the logic to turn on the Blogging Prompts or the analytics tracking, which will be done in separate PRs.

This PR contains some TODOs that I left on purpose since they will be done in the next PRs.

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/79efaa9a-a4b8-4ff0-b0b1-3a06e0f535f1

-----

## To Test:

1. Install and log into the app with a site that has Blogging Prompts available
2. Go to the My Site dashboard
3. Tap "Learn More" on the Bloganuary card
4. **Verify** the overlay modal shows up with the correct copy, depending on whether the current site has prompts enabled or not (test with both)
5. Tap the close button in the Modal
6. **Verify** it closes the modal
7. Open the modal again
8. Tap the CTA action
9. **Verify** it closes the modal (it currently doesn't turn on the prompts for disabled sites)
10. Open the modal again
11. Drag the modal down
12. **Verify** it is draggable and is dismissed when flung down

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

13. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

14. What automated tests I added (or what prevented me from doing so)

    - Added ViewModel unit tests for the modal.

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
